pkg/app/pipedv1/plugin/kubernetes/deployment/canary.go

```diff
@@ -16,18 +16,160 @@ package deployment
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/pipe-cd/pipecd/pkg/app/pipedv1/plugin/kubernetes/config"
+	"github.com/pipe-cd/pipecd/pkg/app/pipedv1/plugin/kubernetes/provider"
+	"github.com/pipe-cd/pipecd/pkg/yamlprocessor"
```

Comment:
Should this yamlprocessor package be part of the SDK as well? 🤔

Reply:
Thanks, now that you mention it, it might be true.

IMO, it should stay as is for now.
It seems this package is mainly for piped logic and is used only for the canary stage as executor logic for now.
WDYT?

Reply:
I see. 

@khanhtc1202 @t-kikuc 
So I think there are two patterns for now. Which is better for you?

1. Add same package to SDK
2. Add same package to k8s plugin 

IMO, 2 is better because it is not expected to be used for other executors for now.

Reply:
We decided to choose 2.

Reply:
Copied on [94ca4e8](https://github.com/pipe-cd/pipecd/pull/5892/commits/94ca4e85d25c2cc720f8ec370f218dc1d9f7b624)